File: code/backend/java/core/src/main/java/games/paths/core/port/story/StoryCatalogExportPort.java
package games.paths.core.port.story;

import games.paths.core.model.story.CatalogExportResult;

/**
 * StoryCatalogExportPort - v0.37.6 inbound port: (re)writes the static catalog
 * files `data/stories-{lang}.json` read by the game home page.
 */
public interface StoryCatalogExportPort {

    /** Relative path of the file for a language. */
    static String pathFor(String lang) {
        return "data/stories-" + lang + ".json";
    }

    /** Writes every configured language. */
    CatalogExportResult exportCatalog();

    /** Thrown when the backend has no export destination configured. */
    class CatalogNotConfiguredException extends RuntimeException {
        public CatalogNotConfiguredException() {
            super("No static catalog destination configured (CATALOG_EXPORT_DIR / WEBSITE_BUCKET)");
        }
    }
}

File: code/backend/python/app/core/ports/story/story_catalog_export_port.py
from abc import ABC, abstractmethod
from app.core.models.story.catalog_export_result import CatalogExportResult


def catalog_path_for(lang: str) -> str:
    return f"data/stories-{lang}.json"


class CatalogNotConfiguredError(RuntimeError):
    def __init__(self):
        super().__init__("No static catalog destination configured (CATALOG_EXPORT_DIR / WEBSITE_BUCKET)")


class StoryCatalogExportPort(ABC):
    """v0.37.6 — inbound port: (re)writes data/stories-{lang}.json for the game home page."""

    @abstractmethod
    def export_catalog(self) -> CatalogExportResult:
        pass
